daemon/exec.go

```diff
@@ -223,11 +227,16 @@ func (daemon *Daemon) ContainerExecStart(ctx context.Context, name string, optio
 		if err != nil {
 			return err
 		}
-		spec := specs.Spec{Process: p}
-		if err := json.Unmarshal(md.Spec.GetValue(), &spec); err != nil {
+
+		// Technically, this should be a [specs.Spec], but we're only
+		// interested in the Process field.
+		spec := &struct{ Process *specs.Process }{p}
+		if err := json.Unmarshal(md.Spec.GetValue(), spec); err != nil {
```

Comment:
IIUC, the intent here is to update fields of `p` according to the JSON content of "process" in `md.Spec.GetValue()` and the `spec` itself is not used at later point.

nit:
```suggestion
		spec := struct{ Process *specs.Process }{p}
		if err := json.Unmarshal(md.Spec.GetValue(), &spec); err != nil {
```

I think this makes more sense - we only need the address for the `json.Unmarshal`, we don't care if `spec` itself is a value or pointer.

Reply:
Ah, yes, agreed; let me update that!

Reply:
applied your changes 👍